### DISCUSSION TITLE
Since Linkerd enabled automatic mTLS between apps, could I just use insecure_channel provided by grpc-python?

### DISCUSSION THREAD

**Discussion by zhoudaxia233** [linkerd/linkerd2#10189] [Q&A] [answered]
I'm using grpc in Python and linkerd to build a demo microservice app. In the python code, I need to create a channel in order to use grpc. Since Linkerd provides mTLS automatically, is an `insecure_channel` sufficient in my application code?

```
>>> import grpc
>>> from recommendations_pb2_grpc import RecommendationsStub
>>> channel = grpc.insecure_channel("localhost:50051")
>>> client = RecommendationsStub(channel)
```

**Comment by kleimkuhler** [comment, marked as answer]
If both client and server are meshed, then yes, the Linkerd proxies on both sides will encrypt the traffic even if the app is using an insecure channel.
